analyses/cell-type-neuroblastoma-04/run-analysis.sh

```diff
@@ -51,32 +75,58 @@ download_file() {
   fi
 }
 
-nbatlas_subset_url="https://data.mendeley.com/public-files/datasets/yhcf6787yp/files/0a569381-3a0c-4eec-863a-e20544b686ed/file_downloaded"
+# Define URLs; the first 2 will always be downloaded
 nbatlas_full_url="https://data.mendeley.com/public-files/datasets/yhcf6787yp/files/f5969395-5f6e-4c5d-a61a-5894773d0fee/file_downloaded"
 nbatlas_tumor_url="https://data.mendeley.com/public-files/datasets/yhcf6787yp/files/78cad1b4-7425-4073-ba09-362ef73c9ab9/file_downloaded"
+nbatlas_subset_url="https://data.mendeley.com/public-files/datasets/yhcf6787yp/files/0a569381-3a0c-4eec-863a-e20544b686ed/file_downloaded"
+
+# Define NBAtlas reference object files with the same name as on Mendeley: https://data.mendeley.com/datasets/yhcf6787yp/3
+nbatlas_full_seurat="${scratch_dir}/seuratObj_NBAtlas_share_v20241203.rds" # this is needed regardless of which nbatlas_version was specified since we need to subset ids
+nbatlas_tumor_metadata_file="${scratch_dir}/SeuratMeta_Share_TumorZoom_NBAtlas_v20250228.rds"
 
-download_file $nbatlas_subset_url $nbatlas_subset_seurat
+# Define which NBAtlas version we're actually converting, and download subset if it's that one
+if [[ $nbatlas_version == "full" ]]; then
+  nbatlas_seurat=${nbatlas_full_seurat}
+else
+  nbatlas_seurat="${scratch_dir}/SeuratObj_Share_50kSubset_NBAtlas_v20240130.rds"
+  download_file $nbatlas_subset_url $nbatlas_seurat
+fi
+
+# grab remaining downloads
 download_file $nbatlas_full_url $nbatlas_full_seurat
 download_file $nbatlas_tumor_url $nbatlas_tumor_metadata_file
 
-# Convert the NBAtlas object to SCE and AnnData formats, for both the full and subsetted versions
-cell_id_file="${scratch_dir}/nbtalast-cell-ids.txt.gz"
+
+# Determine the ids to retain based on the *full* atlas object
+cell_id_file="${scratch_dir}/nbatlas-cell-ids.txt.gz"
 Rscript ${script_dir}/00a_extract-nbatlas-ids.R \
     --nbatlas_file "${nbatlas_full_seurat}" \
     --cell_id_file "${cell_id_file}"
 
+# Convert the NBAtlas object to SCE (and in the future, also to AnnData)
 Rscript ${script_dir}/00b_convert-nbatlas.R \
-    --nbatlas_file "${nbatlas_subset_seurat}" \
-    --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
-    --cell_id_file "${cell_id_file}" \
-    --sce_file "${nbatlas_subset_sce}" \
-    --anndata_file "${nbatlas_subset_anndata}" \
-    ${test_flag}
+   --nbatlas_file "${nbatlas_seurat}" \
+   --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
+   --cell_id_file "${cell_id_file}" \
+   --sce_file "${nbatlas_sce}" \
+   ${test_flag}
+   # For now, we will not save the AnnData object
+   #--anndata_file "${nbatlas_anndata}"
 
-Rscript ${script_dir}/00b_convert-nbatlas.R \
-    --nbatlas_file "${nbatlas_full_seurat}" \
-    --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
-    --cell_id_file "${cell_id_file}" \
-    --sce_file "${nbatlas_full_sce}" \
-    --anndata_file "${nbatlas_full_anndata}" \
-    ${test_flag}
```

Comment:
So I think I would update this to just do the conversion for the full if testing is false and then always do the subset conversion. By default, we want to generate the data for both full and subset, which is not the case currently.